### PR TITLE
Conversion of CSV widget testcases to selenium

### PR DESCRIPTION
**Pull request by radhikav1**
# Convert widgets.CSVWidget.spec.ts test from cypress to selenium

## Description
Conversion of CSV widget testcases to selenium

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-19254](https://cdap.atlassian.net/browse/CDAP-19254)

## Test Plan
Included in e2e tests

## Screenshots
n/a

[CDAP-19254]: https://cdap.atlassian.net/browse/CDAP-19254?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/e2e-test/features/widget.csv.feature [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      When Open Pipeline Studio Page
+      Then Toggle Transform Panel
+      Then Add Projection node to canvas
+      Then Open Projection node properties
+      Then Click add row and enter "value1", "value2" for row 1 and row 2 respectively
+      Then Delete row 2
+      Then Close Projection node properties
+      Then Open Projection node properties
+      Then Verify CSV Widget values "value1", "value2" in Pipeline Stage JSON
+      Then Exit Studio Page
```
Comment (seanfeld) [inline src/e2e-test/features/widget.csv.feature]: new line

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+
+public class WidgetCSV {
+//    private NodeInfo projectionNode = new NodeInfo("Projection", "transform", "0");
+    private String sectionName = "drop";
+
```
Comment (seanfeld) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java:29]: why are there so many empty lines here?

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+
+public class WidgetCSV {
+//    private NodeInfo projectionNode = new NodeInfo("Projection", "transform", "0");
```
Comment (seanfeld) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: is this comment necessary?

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    private String sectionName = "drop";
+
+
+
+    @Then("Verify that the CSV Widget contains only one row by default")
+    public void verifyDefaultCSVWidget() {
+        Assert.assertTrue(
```
Comment (seanfeld) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: Please reduce the tab or space width - should be 2 like the rest of the code

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                Helper.isElementExists(
+                        By.xpath("//*[@data-testid='" + sectionName + "']//*[@data-testid='0']")));
```
Comment (seanfeld) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: why are all of these xpath?

**Review by seanfeld — CHANGES_REQUESTED** [review]
mainly nits

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java (hunk 5 above)
Comment (radhikav1) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: There is no unique datatestid with which we can get the required element as Widget component has repeated structure(like rows) .  By using complete Xpath only we are able to get the required element, so has to used this way . 

I can check on making  these repeated Xpaths as paramaterized methods

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java (hunk 2 above)
Comment (radhikav1) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java:29]: Refactoring mistake. corrected now.

File: src/e2e-test/features/widget.csv.feature (hunk 1 above)
Comment (radhikav1) [reply src/e2e-test/features/widget.csv.feature]: done

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java (hunk 3 above)
Comment (radhikav1) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: Not required, removed now.

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java (hunk 4 above)
Comment (radhikav1) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: Changed now

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java (hunk 5 above)
Comment (radhikav1) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCSV.java]: Changed Xpaths to datatest-Id APIs and also refactored the code as possible

**Review by seanfeld — APPROVED** [review]

**Review by sumengwang — APPROVED** [review]

**Pull request merged**